datastream-kafka/src/main/java/com/linkedin/datastream/kafka/KafkaTransportProvider.java

```diff
@@ -118,8 +118,14 @@ public KafkaTransportProvider(DatastreamTask datastreamTask, List<KafkaProducerW
     } else {
       // If the partition is not specified. We use the partitionKey as the key. Kafka will use the hash of that
       // to determine the partition. If partitionKey does not exist, use the key value.
-      keyValue = record.getPartitionKey().isPresent() ? record.getPartitionKey().get().getBytes() : keyValue;
-      return new ProducerRecord<>(topicName, keyValue, payloadValue);
+      ProducerRecord<byte[], byte[]> producerRecord = new ProducerRecord<>(topicName, payloadValue);
```

Comment:
To simplify the change, you could just modify the original code.

Old: 
keyValue = record.getPartitionKey().isPresent() ? record.getPartitionKey().get().getBytes() : keyValue;
New: 
keyValue = record.getPartitionKey().isPresent() ? record.getPartitionKey().get().getBytes() : null;

Or you can set byte[] keyValue = null on line 100.